core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala

```diff
@@ -141,21 +145,30 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit F: Async[F])
     */
   def send(record: ProducerRecord[K, V]): F[F[RecordMetadata]] =
     // inspired by https://github.com/fd4s/fs2-kafka/blob/series/3.x/modules/core/src/main/scala/fs2/kafka/KafkaProducer.scala
-    F.delay(Promise[RecordMetadata]())
-      .flatMap { promise =>
-        sendRaw2(record, promise.complete)
-          .map(cancel =>
-            F.fromFutureCancelable(
-              F.delay((promise.future, cancel))
-            )
-          )
-      }
+    for {
+      promise <- F.delay(Promise[RecordMetadata]())
+      sent <- Deferred[F, Unit]
+      cancelToken <- Deferred[F, F[Unit]]
+      _ <- supervisor.supervise(
+        T.spanR("kafka4s-send").use(wrap =>
+          for {
+            cancel <- wrap(T.span("buffer")(sendRaw2(record, promise.complete)))
```

Comment:
We want everything to happen in `spanR`, but the cancel token needs to be exposed outside.